File: C/pset2/readability.c
#include <ctype.h>
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <math.h>

//index = 0.0588 * length - 0.296 * sentences - 15.8;

// define variables
int spaces;
int sentences;
int other;


int main(void)
{
    string text = get_string("Text: ");




// compute the information about text
    for (int i = 0, n = strlen(text); i < n; i++)
    {

// compute the spaces in text
        if (text[i] == 32)
        {
            spaces += 1;
        }
        if (text[i] == 46 || text[i] == 33 || text[i] == 63)
        {
            sentences += 1;
        }
        if (text[i] == 34 || text[i] == 39 || text[i] == 44)
        {
            other += 1;
        }
    }
// define letters and words
    float letters = strlen(text) - spaces - sentences - other;
    float words = spaces + 1;

// compute pr 100
    float s = sentences * 100 / words;
    float l = letters * 100 / words;

// compute the grade

    int grade = round(0.0588 * l - 0.296 * s - 15.8);
    if (grade < 1)
    {
        printf("Before Grade 1\n");
        return 0;
    }
    if (grade > 16)
    {
        printf("Grade 16+\n");
        return 0;
    }
    else
    {
        printf("Grade %d\n", grade);
    }

}

File: Python/pset6/readability.py
from cs50 import get_string


def main():
    
    # get text from user
    text = get_string("Text:")

    # initialise variables
    spaces = 0
    sentences = 0
    other = 0

    # calculate the different values
    for i in range(0, len(text)):
        if ord(text[i]) == 32:
            spaces += 1
        elif ord(text[i]) == 46 or ord(text[i]) == 33 or ord(text[i]) == 63:
            sentences += 1
        elif ord(text[i]) == 34 or ord(text[i]) == 39 or ord(text[i]) == 44:
            other += 1

    # define letters and words
    words = spaces + 1
    letters = len(text) - spaces - sentences - other

    # compute pr 100
    s = sentences * 100 / words
    l = letters * 100 / words

    # comute the grade
    grade = round(0.0588 * l - 0.296 * s - 15.8)

    if grade < 1:
        print("Before Grade 1")

    elif grade > 16:
        print("Grade 16+")

    else:
        print("Grade", grade)


if __name__ == "__main__":
    main()